pptx/text/bullets.py

```diff
@@ -0,0 +1,720 @@
+        elif isinstance(xBulletFont, CT_TextFont):
+            bullet_font_cls = _BulletFontSpecific
+        else:
+            bullet_font_cls = _BulletFont
+        
+        return super(_BulletFont, cls).__new__(bullet_font_cls)
+        
+    @property
+    def type(self):
+        return "NoBulletFont"
+
+    @property
+    def typeface(self):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletFont type %s has no typeface property, call .set_typeface() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @typeface.setter
+    def typeface(self, value):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletFont type %s has no typeface property, call .set_typeface() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+    
+    @property
+    def pitch_family(self):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletFont type %s has no pitch_family property, call .set_typeface() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @pitch_family.setter
+    def pitch_family(self, value):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletFont type %s has no pitch_family property, call .set_typeface() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+
+    @property
+    def panose(self):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletFont type %s has no panose property, call .set_typeface() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+    
+    @panose.setter
+    def panose(self, value):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletFont type %s has no panose property, call .set_typeface() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+    
+    @property
+    def charset(self):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletFont type %s has no charset property, call .set_typeface() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+    
+    @charset.setter
+    def charset(self, value):
+        """Raise TypeError for types that do not override this property."""
+        tmpl = (
+            "BulletFont type %s has no charset property, call .set_typeface() first"
+        )
+        raise TypeError(tmpl % self.__class__.__name__)
+
+
+class _BulletFontFollowText(_BulletFont):
+    """
+    Designates that the Bullet Font will match the accompanying paragraph text.
+    """
+    @property
+    def type(self):
+        return "BulletFontFollowText"
+
+class _BulletFontSpecific(_BulletFont):
```

Comment:
This class looks very similar to `BulletFont` on line 511. I'm wondering if we could get rid of `_BulletFontSpecific` and `_BulletFontFollowText` and we make the bullet font to follow the paragraph text by default and still allow for the user to change the properties.

Reply:
Unfortunately, we need to follow the XML schema as most of this is predefined by powerpoint and the library follows the schema very closely.